### ISSUE TITLE
too much recursion

### ISSUE THREAD

**Issue by bmcage** [Spiritdude/OpenJSCAD.org#63]
On an example with simple lasercut box, you quickly hit 'too much recursion'
Unclear why this would be complicated. 

Is there a way to program around this limitation. As the aim is lasercutting, having lines which should not be lasercut is a not acceptable, so I need a full side, but for a box, I will need 6 sides.

Test file that hits this limitation: https://gist.github.com/bmcage/a6c6b234911971bc390a 
To make it work: at the bottom of this file, comment out two of four the for loops

At the top of the file you see:

    var shape1 = new CSG.Polygon2D( polypoints ).extrude({offset: [0, 0, thick]});
    //var shape1 = CAG.fromPoints( polypoints );

Both fail. Idea is to show the box nicely and allow dxf output for lasercutting

**Comment by bmcage** [comment]
Further investigation shows this is due to end point and begin point of polygon being equal.

Simple test case to see this: 
```javascript
function main() {    
    var points = [new CSG.Vector2D(0,0)];
    points[1] = new CSG.Vector2D(0,10);
    points[2] = new CSG.Vector2D(10,10);
    points[3] = new CSG.Vector2D(0,0);
    var shape1 = CAG.fromPoints( points );
   return shape1;
}
```

At a minimum, a clear error to the user when begin and endpoint are equal would be nicest. Allowing begin=end would nicer still!

**Comment by bmcage** [comment]
test in openscad shows that begin=end is allowed there. Just paste:
```
polygon( points=[[0,0],[2,1],[1,2],[1,3],[3,4],[0,5],[0,0] ]);
```
and press F5 and F6.

**Comment by richievos** [comment]
@bmcage could you submit a pull request making that compatible with OpenSCAD's way of doing things? I haven't personally run into this, but seems like a quick fix.

**Comment by bmcage** [comment]
@richievos I don't have a patch, so not sure what you mean with a pull request. I would expect opening an issue to be the only step possible at the moment.

You hit the same bug when internal points are double. I have that in some procedurs for simplicity of code, eg a kerf on a lasercutter. If you set kerf=0, my procedures make polygons with twice the same point, which works in openscad, but not in openJSCAD, For openJSCAD, I need to add if then blocks testing on the 0.

**Comment by bebbi** [comment]
fixed in joostn/OpenJsCad#54
See also #33

**Comment by z3dev** [comment]
I tested the examples from @bmcage using Release 0.3.0. Both render properly. I'm closing this issue.
